app/src/androidTest/java/org/sagebionetworks/research/mpower/RoomScheduledActivityTests.kt

```diff
@@ -0,0 +1,239 @@
+
+import android.arch.persistence.room.Room
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertNull
+import junit.framework.Assert.assertTrue
+import org.joda.time.DateTime
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.sagebionetworks.bridge.rest.RestUtils
+import org.sagebionetworks.bridge.rest.model.ScheduleStatus
+import org.sagebionetworks.bridge.rest.model.ScheduledActivityListV4
+import org.sagebionetworks.research.mpower.room.ResearchDatabase
+import org.sagebionetworks.research.mpower.room.RoomScheduledActivity
+import org.sagebionetworks.research.mpower.room.RoomScheduledActivityDao
+import org.sagebionetworks.research.mpower.room.RoomTypeConverters
+import java.io.IOException
+import java.nio.charset.Charset
+
+@RunWith(AndroidJUnit4::class)
+class RoomScheduledActivityTests {
+
+    private var database: ResearchDatabase? = null
+    private var activityDao: RoomScheduledActivityDao? = null
+    private var activityList: ScheduledActivityListV4? = null
+    private var roomActivityList: List<RoomScheduledActivity>? = null
+
+    @Before @Test
+    fun setupRoom() {
+        database = Room.inMemoryDatabaseBuilder(
+                InstrumentationRegistry.getTargetContext(), ResearchDatabase::class.java)
+                .allowMainThreadQueries().build()
+        assertNotNull(database)
+
+        activityDao = database?.activitiesDao()
+        assertNotNull(activityDao)
+
+        activityList = RestUtils.GSON.fromJson(
+                resourceAsString("mpower_activities.json"),
+                ScheduledActivityListV4::class.java)
+        assertNotNull(activityList)
+        assertEquals(6, activityList?.items?.size)
+
+        roomActivityList = RoomTypeConverters().fromScheduledActivityListV4(activityList)
+        assertNotNull(roomActivityList)
+        assertEquals(6, roomActivityList?.size)
+    }
+
+    @After
+    fun tearDownRoom() {
+        database?.close()
+    }
+
+    @Test
+    fun serialization_testTaskReference() {
+        assertMedicationTaskReferenceActivity(roomActivityList?.first())
+    }
+
+    fun assertMedicationTaskReferenceActivity(activity: RoomScheduledActivity?) {
+        assertNotNull(activity)
+        assertEquals("273c4518-7cb6-4496-b1dd-c0b5bf291b09:2018-08-17T00:00:00.000", activity?.guid)
+        assertEquals("e24e6601-1822-48b0-8770-00870d870708", activity?.schedulePlanGuid)
+        assertEquals(false, activity?.persistent)
+        assertNotNull(activity?.scheduledOn)
+        assertEquals(DateTime.parse("2018-08-17T00:00-04:00").toDate().time, activity?.scheduledOn?.toDate()?.time)
+        assertNotNull(activity?.expiresOn)
+        assertEquals(DateTime.parse("2018-08-18T00:00-04:00").toDate().time, activity?.expiresOn?.toDate()?.time)
```

Comment:
Does this also test timezones? How important is it that we make sure timezones are being parsed correctly? How difficult would it be to test that?